spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/LogisticRegressionClassifier.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.spark.mllib.feature.HashingTF;
+import org.apache.twill.filesystem.Location;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * SparkCompute that uses a trained Logistic Regression model to classify and tag input records.
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(LogisticRegressionClassifier.PLUGIN_NAME)
+@Description("Uses a Logistic Regression model to classify records.")
+public class LogisticRegressionClassifier extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  public static final String PLUGIN_NAME = "LogisticRegressionClassifier";
+  private final Config config;
+  private Schema outputSchema;
+
+  private LogisticRegressionModel loadedModel = null;
+  private HashingTF tf = null;
+
+  public LogisticRegressionClassifier(Config config) {
+    this.config = config;
+  }
+
+  /**
+   * Configuration for the LogisticRegressionClassifier.
+   */
+  public static class Config extends PluginConfig {
+
+    @Description("The name of the FileSet to load the model from.")
+    private String fileSetName;
+
+    @Description("Path of the FileSet to load the model from.")
+    private String path;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be used for classification.")
+    private String featureFieldsToInclude;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be excluded from being used in classification.")
+    private String featureFieldsToExclude;
+
+    @Description("The field on which to set the prediction. It will be of type double.")
+    private String predictionField;
+
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
+                  " number of features used in LogisticRegressionTrainer. The default value if none is provided " +
+                  " will be 100.")
+    private Integer numFeatures;
+
+    public Config() {
+      this.numFeatures = 100;
+    }
+
+    public Config(String fileSetName, String path, @Nullable String featureFieldsToInclude,
+                  @Nullable String featureFieldsToExclude,
+                  String predictionField, @Nullable Integer numFeatures) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featureFieldsToInclude = featureFieldsToInclude;
+      this.featureFieldsToExclude = featureFieldsToExclude;
+      this.predictionField = predictionField;
+      this.numFeatures = numFeatures;
+    }
+
+    private void validate(Schema inputSchema) {
+      Schema.Field predictionField = inputSchema.getField(this.predictionField);
+      Preconditions.checkArgument(predictionField == null, "Prediction field must not already exist in input schema.");
+
+      SparkUtils.validateConfigParameters(inputSchema, featureFieldsToInclude, featureFieldsToExclude,
```

Comment:
It would be better if there was a concrete class that the configs extended rather than a utils class. For example, MLConfig, which contains all common properties and a validate() method.